overlord/ifacestate/handlers.go

```diff
@@ -56,6 +56,10 @@ func (m *InterfaceManager) doSetupProfiles(task *state.Task, _ *tomb.Tomb) error
 	// but remember the old value in the task in case we undo.
 	task.Set("old-devmode", snapState.DevMode())
 	snapState.SetDevMode(ss.DevMode())
+
+	task.Set("old-confined", snapState.Confined())
```

Comment:
I'm wondering if we actually need this as a persistent flag here. Isn't confined just !devmode at this point?